### ISSUE TITLE
EFI support and install fails (Bugzilla Bug 465)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#465]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 465
Date: 2014-01-28 10:23:39 +0000
From: Efrem McCrimon &lt;<efremmc2@gmail.com>&gt;
To:   OpenMandriva QA &lt;<bugs@openmandriva.org>&gt;
CC:   @jclvanier, jeremielapuree@yahoo.fr, @tpgxyz

Duplicates:   #568
See also:     https://issues.openmandriva.org/show_bug.cgi?id=568
Last updated: 2014-03-13 15:08:26 +0000

## Comment 3031
Date: 2014-01-28 10:23:39 +0000
From: Efrem McCrimon &lt;<efremmc2@gmail.com>&gt;

grub2 install:

root@localhost live]# grub2-install --efi-directory=/boot/efi/EFI/ /dev/sda2
Path ‘/boot/grub2’ is not readable by GRUB on boot. Installation is impossible. Aborting.
[root@localhost live]#

///

root@localhost live]# grub2-install --recheck --boot-directory=/boot/efi/EFI/ --efi-directory=/boot/efi/EFI/ /dev/sda2
/usr/sbin/grub2-bios-setup: warning: File system ‘fat’ doesn't support embedding.
/usr/sbin/grub2-bios-setup: warning: Embedding is not possible.  GRUB can only be installed in this setup by using blocklists.  However, blocklists are UNRELIABLE and their use is discouraged..
/usr/sbin/grub2-bios-setup: error: will not proceed with blocklists.
[root@localhost live]# efibootmgr -v
modprobe: FATAL: Module efivars not found.
Fatal: Couldn't open either sysfs or procfs directories for accessing EFI variables.
Try 'modprobe efivars' as root.
[root@localhost live]# modprobe efivars
modprobe: FATAL: Module efivars not found.
[root@localhost live]# grub2-install --recheck --boot-directory=/boot/efi/EFI/ --efi-directory=/boot/efi/EFI/ /dev/sda2
/usr/sbin/grub2-bios-setup: warning: File system ‘fat’ doesn't support embedding.
/usr/sbin/grub2-bios-setup: warning: Embedding is not possible.  GRUB can only be installed in this setup by using blocklists.  However, blocklists are UNRELIABLE and their use is discouraged..
/usr/sbin/grub2-bios-setup: error: will not proceed with blocklists.
[root@localhost live]# man blocklists
No manual entry for blocklists

///
root@localhost live]# blkid | grep EFI
/dev/sda2: LABEL="SYSTEM_DRV" UUID="C8E8-9624" TYPE="vfat" PARTLABEL="EFI system partition" PARTUUID="9da2222f-1b29-482c-87da-ce6794df8568"
[root@localhost live]#

=====================

blkid reports:
 partition" PARTUUID="9da2222f-1b29-482c-87da-ce6794df8568"
[root@localhost live]# blkid
/dev/sdb1: LABEL="OPENMANDRIV" UUID="8EEE-EF43" TYPE="vfat"
/dev/sda1: LABEL="WINRE_DRV" UUID="8EE4E677E4E66147" TYPE="ntfs" PARTLABEL="Basic data partition" PARTUUID="8786cf68-ea2e-4e20-9069-798b57a1ecb4"
/dev/sda2: LABEL="SYSTEM_DRV" UUID="C8E8-9624" TYPE="vfat" PARTLABEL="EFI system partition" PARTUUID="9da2222f-1b29-482c-87da-ce6794df8568"
/dev/sda3: LABEL="LRS_ESP" UUID="42EC-3017" TYPE="vfat" PARTLABEL="Basic data partition" PARTUUID="e8f0e2e7-ed91-4bf6-b425-cb5efc38cbd2"
/dev/sda5: LABEL="Windows8_OS" UUID="9440EEC940EEB0E2" TYPE="ntfs" PARTLABEL="Basic data partition" PARTUUID="6a45d4db-03af-49e4-ab56-4d4d7c779783"
/dev/sda6: UUID="fe8bf7e2-8e5b-4464-82b7-e1015f0b3895" TYPE="ext4" PARTUUID="87ee8ecc-c645-4d14-9e02-5e43035351f4"
/dev/sda7: LABEL="LENOVO" UUID="968416E48416C721" TYPE="ntfs" PARTLABEL="Basic data partition" PARTUUID="f0a13edf-68bf-40ac-9f85-1969dc4310a8"
/dev/sda8: LABEL="PBR_DRV" UUID="4C0CF2310CF215A4" TYPE="ntfs" PARTLABEL="Basic data partition" PARTUUID="16db5139-a9c3-4775-b290-7ab474222cba"
/dev/sda9: UUID="692272f5-3c06-461d-9195-98ba740fcf58" TYPE="swap" PARTUUID="41688099-3abd-4df1-9112-1b2e374005ab"
/dev/sda10: UUID="66094a1b-8b0d-45d6-a754-b5a67bd54a2a" TYPE="ext4" PARTUUID="1bd25656-4e90-4301-a5c6-441acbc1fc4c"
/dev/loop0: TYPE="squashfs"
/dev/loop1: TYPE="DM_snapshot_cow"
/dev/loop2: TYPE="squashfs"
/dev/loop3: LABEL="_OpenMandriva.20" UUID="4bee204e-b524-453e-b128-9b140d4eb7a7" SEC_TYPE="ext2" TYPE="ext3"
/dev/loop4: TYPE="DM_snapshot_cow"
/dev/mapper/live-rw: LABEL="_OpenMandriva.20" UUID="4bee204e-b524-453e-b128-9b140d4eb7a7" TYPE="ext3"
/dev/sda4: PARTLABEL="Microsoft reserved partition" PARTUUID="7224892c-f93f-4cc9-84b1-a4a8b791b422"
[root@localhost live]#

===uname erpoerts:

 
[root@localhost live]# uname -a
Linux localhost 3.11.8-nrjQL-desktop-1omv #1 SMP PREEMPT Tue Nov 19 21:38:07 UTC 2013 x86_64 x86_64 x86_64 GNU/Linux
[root@localhost live]#

=================

Boot from USB 16GB drive - to LiveUSB

 2013 x86_64 x86_64 x86_64 GNU/Linux
[root@localhost live]# fdisk -l
WARNING: fdisk GPT support is currently new, and therefore in an experimental phase. Use at your own discretion.

Disk /dev/sda: 256.1 GB, 256060514304 bytes, 500118192 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk label type: gpt

#         Start          End    Size  Type            Name
 1         2048      2050047   1000M  Windows recover Basic data partition
 2      2050048      2582527    260M  EFI System      EFI system partition
 3      2582528      4630527   1000M  unknown         Basic data partition
 4      4630528      4892671    128M  Microsoft reser Microsoft reserved partition
 5      4892672    299483135  140.5G  Microsoft basic Basic data partition
 6    299483136    349814783     24G  Microsoft basic
 7    467255296    475643903      4G  Microsoft basic Basic data partition
 8    475643904    500117503   11.7G  Windows recover Basic data partition
 9    349814784    374980607     12G  Microsoft basic
10    374980608    467255295     44G  Microsoft basic

Disk /dev/sdb: 16.0 GB, 16008609792 bytes, 31266816 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disk label type: dos
Disk identifier: 0x0001bf82

   Device Boot      Start         End      Blocks   Id  System
/dev/sdb1   *        2048    31266815    15632384    c  W95 FAT32 (LBA)

Disk /dev/mapper/live-rw: 9114 MB, 9114222592 bytes, 17801216 sectors
Units = sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes

[root@localhost live]#

(My fdisk report)
This machine is a Yoga Pro 2 - i7, 256GB SSD,
cat /etc/release
OpenMandriva Lx release 2013.0 (Oxygen) for x86_64

## Comment 3033
Date: 2014-01-28 10:42:53 +0000
From: @jclvanier

Suggestions:
-- in you install grub2 from the the live cd, you should mount the "real" boot partition first and adapt the options --efi-directory= and boot-directory= accordingly
-- try with the option --force, if you are ready to ignore the warning
-- use grub-install (from the package grub-efi, already installed)

## Comment 3045
Date: 2014-01-28 16:04:05 +0000
From: Efrem McCrimon &lt;<efremmc2@gmail.com>&gt;

First, i mounted the boot drive:

mount /dev/sda2 /boot/efi

Then

I used: 
grub2-install --recheck --boot-directory=/boot/efi/EFI/ --efi-directory=/boot/efi/EFI/ --force /dev/sda2

It did the install. After reboot, I removed the USB-stick and I get the error "no bootable device --insert boot disk" because it tries to boot legacy.

I had to change my bios setting to first turn-off secure boot, uefi,and fastboot
I enabled - legacy support and boot legacy first = without these options i could not boot from the usb-drive with the live usb.  The usb-drive for this distribution would not boot with uefi enabled.  It boots a blank screen without giving me a chance to change the resolution and I noticed that the usb-drive blinks.

Is there a newer version of the iso for 64-bit?  Or should I switch to something else?  See below.

## Comment 3046
Date: 2014-01-28 16:33:58 +0000
From: Efrem McCrimon &lt;<efremmc2@gmail.com>&gt;

Does this 2013 version support UEFI mode?  Why does the usb-stick boot to a blank screen without having the option to select an option.  Can I alter the usb-stick configuration so that it will boot?  I have Windows 8.1 installed on a Lenovo Yoga Pro 2 with a 256Gb SSD drive.  I partition it to have 80GB for Linux and craved it up as: 12GB for / (root), 12GB (swap space-plan for hiberation), ~48Gb for /home.

What do you recommend for to get this to work?
I am surprise that I ma having problems with this.  In legacy boot, I had to change the video uvesafb option 1280x1024 (otherwise I could not see anything) - it does not boot on 1024x768-16 or -32.  This system has a weird video card (3200x1800 resolution - 13.3in screen)

## Comment 3048
Date: 2014-01-29 03:39:37 +0000
From: Efrem McCrimon &lt;<efremmc2@gmail.com>&gt;

I am waiting on what action should I take?  I have another computer using Mandriva 2012 that I use as my main development system.

## Comment 3128
Date: 2014-02-06 06:33:46 +0000
From: jeremielapuree@yahoo.fr

What you see, it is the infamous bug #390.
Your only hope: se a another distrib. Openmandriva does not support UEFI yet.
Or install in legacy mode. But then, you can not have a dual boot with windows ifif windows in installed in UEFI mode.....

## Comment 3882
Date: 2014-03-13 15:02:45 +0000
From: @tpgxyz

I'm closing this bug as a duplicate of bug #568 a one bug to cover missing UEFI support in OMV Lx.

TPG
Cooker Team

*** This bug has been marked as a duplicate of bug #568 ***
